js/globals.ts

```diff
@@ -52,6 +52,7 @@ Object.freeze(window.Deno);
 // by ObjectCreate(%ObjectPrototype%), instead of %ObjectPrototype%.
 let console = Object.create({}) as consoleTypes.Console;
 Object.assign(console, new consoleTypes.Console(core.print));
+console[consoleTypes.isConsoleInstance] = true;
```

Comment:
@justjavac could you explain why this is needed if it's already done in `Console` constructor? I've started to work on #2521 but hit the wall with that property

Reply:
ref https://console.spec.whatwg.org/#console-namespace

> For historical web-compatibility reasons, the namespace object for console must have as its [[Prototype]] an empty object, created as if by `ObjectCreate(%ObjectPrototype%)`, instead of `%ObjectPrototype%`.

If we use the `Console` constructor to create a `console` instance, we will not follow the specification.

also see: https://github.com/justjavac/deno/blob/dfb609bc469a1ac365be0526d411337f2c66135e/js/console_test.ts#L15-L21

Reply:
Alright, thanks for explanation. Still, can we somehow circumvent that and don't require to manually set `console[consoleTypes.isConsoleInstance] = true;` but instead do it automagically?

I'm asking because I'd like to expose console class - that way in testing framework we'd be able to create console with print functions that captures all output (needed for "silent" mode as well as snapshot tests).

Reply:
Thanks @justjavac!